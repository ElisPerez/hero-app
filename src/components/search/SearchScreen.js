import { useForm } from "../../hooks/useForm";
import { getHeroByName } from "../../selectors/getHeroByName";
import { HeroCard } from "../hero/HeroCard";


export const SearchScreen = () => {

 
 const [ formValues, handleInputChange ] = useForm({
  searchText: '',
 });

 const { searchText } = formValues;

 const heroesFilted = getHeroByName('spiderman');

 const handleSearch = (e) => {
  e.preventDefault();
  console.log(searchText)

  
 }

 return (
  <div>
   <h1>Search Screen</h1>
   <hr />
   <div className="row">
    <div className="col-5">
     <h4>Search</h4>
     <hr />

     <form onSubmit={handleSearch}>
      <input
       type="text"
       placeholder="Search hero"
       className="form-control"
       name="searchText"
       autoComplete="off"
       value={ searchText }
       onChange={ handleInputChange }
      />
      <button
       type="submit"
       className="btn btn-outline-light mt-3"
      >
       Search
      </button>
     </form>
    </div>

    <div className="col-7">
     <h4>Resultados</h4>
     <hr />

     {
      heroesFilted.map(hero => (
       <HeroCard
        key={ hero.id }
        {...hero}
       />
      ))
     }
    </div>
   </div>
  </div>
 )
}