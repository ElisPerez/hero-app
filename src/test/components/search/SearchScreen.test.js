import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../../components/search/SearchScreen";


describe('Pruebas en <SearchScreen />', () => {
 test('debe de mostrarse correctamente con valores por defecto', () => {
  const wrapper = mount(
   <MemoryRouter initialEntries={['/search']}>
    <SearchScreen />
   </MemoryRouter> 
    );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.alert').text().trim()).toBe('The results will be displayed here...');
 }); // cierre test 1
 
}); // cierre describe
