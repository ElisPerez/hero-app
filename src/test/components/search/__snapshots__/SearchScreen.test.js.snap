// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-light mt-3"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert bg-dark"
      >
        The results will be displayed here...
      </div>
      <div
        className="row row-cols-1 g-3"
      />
    </div>
  </div>
</div>
`;
